File: PizzeriaRestApi/src/main/java/rest/api/entities/Commande.java
package rest.api.entities;
import java.util.ArrayList;
import java.util.Date;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.*;


@Entity
@Data
@NoArgsConstructor 
@AllArgsConstructor
public class Commande {
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	
	private int id_com;
	private Date date_com;
	private String etat;
	private Integer prix;
	private String type;
	private String adresse;
	
	

	 @ManyToOne private Client client;
	 @JsonIgnore
	@OneToMany(mappedBy="commande",cascade={CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REMOVE
        })
	
	private List<Lignedecommande> lignedecommandes;


	
	
	

}

File: PIZZERIA-MOBILE/lib/configuration/commande.dart
import 'package:food_order_ui/configuration/artcile.dart';
import 'package:food_order_ui/configuration/client.dart';
import 'package:intl/intl.dart';

class Commande {
  int id_com;
  String adresse;
  String date_com;
  String etat;
  int prix;
  String type;
  Client client;

  Commande(this.id_com, this.adresse, this.date_com, this.etat, this.prix,
      this.type, this.client);

  factory Commande.fromJson(Map<String, dynamic> json) {
    return new Commande(
      json["id_com"],
      json["adresse"],
      json["date_com"],
      json["etat"],
      json["prix"],
      json["type"],
      Client.fromJson(json["client"]),
    );
  }
}
